### ISSUE TITLE
bug renaming, deleting substrates effect on auto-update to Rules, etc.

### ISSUE THREAD

**Issue by rheiland** [PhysiCell-Tools/PhysiCell-Studio#152]
There is a fatal bug with renaming/deleting substrates and the effect it has on the Rules comboboxes. A minimal test *seems* to be: start with the template.xml (where there's a single substrate called `substrate`), make a Copy (`substrate01`), rename the original to be `sub`, delete the copy.
```
-------- rules_tab.py: delete_substrate(), self.substrates = ['sub', 'sub01']
Traceback (most recent call last):
  File "/Users/heiland/git/studio_dev/bin/microenv_tab.py", line 762, in delete_substrate
    self.rules_tab.delete_substrate(self.current_substrate)
  File "/Users/heiland/git/studio_dev/bin/rules_tab.py", line 890, in delete_substrate
    self.substrates.remove(name)
ValueError: list.remove(x): x not in list
Abort trap: 6
```

It seems pretty obvious what's happening: erroneous string substitution.

Related, *somehow* through a sequence of copy/delete substrates, I was able to get duplicate named `substrate01`, `substrate01`. Can't reproduce right now.
<img width="316" alt="Screenshot 2023-09-19 at 4 24 56 AM" src="https://github.com/PhysiCell-Tools/PhysiCell-Studio/assets/1774917/9077bbe0-100f-42a3-80f9-77650e98d5a5">

**Comment by drbergman** [comment]
On the duplication, I've gotten that when I've closed and reopened studio. It boots with the index reset to 1 so that the first new substrate is always called substrate01 until you rename it. I had explored a solution, but figured this was intentional (since cell defs already do some randomization) and argued that users can reasonably be held responsible for giving their substrates workable names. I can make a PR on that soon
